filebrowser_safe/views.py

```diff
@@ -173,30 +180,47 @@ def browse(request):
                     results_var["select_total"] += 1
             except OSError:
                 # Ignore items that have problems
-                continue
+                return
             else:
-                files.append(fileobject)
+                sort_values[fileobject.filename] = getattr(fileobject, query["o"])
+                files_to_sort.append(fileobject)
                 results_var["results_current"] += 1
 
-        # COUNTER/RESULTS
-        if fileobject.filetype:
-            counter[fileobject.filetype] += 1
+    for i in range(0, file_list_size, batch_size):
+        file_list = file_list[i : i + batch_size]
+        for file in file_list:
+            # EXCLUDE FILES MATCHING ANY OF THE EXCLUDE PATTERNS
+            filtered = not file or file.startswith(".")
+            for re_prefix in filter_re:
+                if re_prefix.search(file):
+                    filtered = True
+            if filtered:
+                continue
+            proc_file = threading.Thread(
+                target=create_file_object,
+                args=(
+                    file,
+                    files_to_sort,
+                ),
+            )
+            proc_file.start()
+            proc_list.append(proc_file)
+        for fo in proc_list:
+            fo.join()
 
     # SORTING
-    query["o"] = request.GET.get("o", fb_settings.DEFAULT_SORTING_BY)
-    query["ot"] = request.GET.get("ot", fb_settings.DEFAULT_SORTING_ORDER)
-    defaultValue = ""
-    if query["o"] in ["date", "filesize"]:
-        defaultValue = 0.0
-    files = sorted(files, key=lambda f: getattr(f, query["o"]) or defaultValue)
+    files_to_sort = sorted(
+        files_to_sort, key=lambda f: sort_values[f.filename] or defaultValue
+    )
```

Comment:
❔ Question only for my own curiosity -- is the idea here that using dict `sort_values` to access the value is faster for `sorted` to work with (as compared to the list `files` from before)?

Reply:
it is faster, but not saving any additional time with this change. it's updated to this because I moved get_attr(...) into the loop that was happening earlier (so it can be apart of the threaded calls) and just stored the values in a map for use here.

Reply:
Ah gotcha, I see now.  Thanks!